tests/core/fsck-on-boot/task.yaml

```diff
@@ -47,22 +47,19 @@ execute: |
       # unset RecoverySystem in the modeenv, and it's not necessary for
       # seeding anymore
 
-      # Refer to the core 20 gadgets for details:
-      # https://github.com/snapcore/pc-amd64-gadget/blob/20/gadget.yaml
-      # https://github.com/snapcore/pi-gadget/blob/20-arm64/gadget.yaml
       if mountpoint /run/mnt/snapd >/dev/null; then
-        if [[ "$SPREAD_SYSTEM" == ubuntu-core-20-arm-* ]]; then
           umount /run/mnt/ubuntu-seed/snaps/snapd_*.snap
-        else
-          umount /run/mnt/ubuntu-seed/systems/*/snaps/snapd_*.snap
-        fi
       fi
       umount /var/lib/snapd/seed
       umount /run/mnt/ubuntu-seed
-      if [[ "$SPREAD_SYSTEM" == ubuntu-core-20-arm-* ]]; then
-        umount /boot/uboot
+
+      # Refer to the core 20 gadgets for details:
+      # https://github.com/snapcore/pc-amd64-gadget/blob/20/gadget.yaml
+      # https://github.com/snapcore/pi-gadget/blob/20-arm64/gadget.yaml
+      if os.query is-arm; then
+          umount /boot/uboot
```

Comment:
The rest of the test looks to be using 2 spaces for indentation.

Reply:
updated